ResearchKit/ActiveTasks/ORKWorkoutConnector.swift

```diff
@@ -148,17 +148,17 @@ open class ORKWorkoutConnector: NSObject, HKWorkoutSessionDelegate, WCSessionDel
     public var connectivitySession: WCSession? {
         return _connectivitySession
     }
-    private var _connectivitySession: WCSession?
+    fileprivate var _connectivitySession: WCSession?
     
     /**
      Should the connector send messages to the phone?
      */
     public var startedFromPhone = false
     
     /**
-     Should the connector timeout after a given duration?
+     Should the connector timeout after a given duration? Default = 15 minutes.
      */
-    public var workoutDuration: TimeInterval = 0 {
+    public var workoutDuration: TimeInterval = 15.0 * 60.0 {
         didSet {
             if workoutDuration > 0 && timer == nil && _workoutSession != nil {
```

Comment:
No, the timer just looks at "when did the workout session start?". It's a pair of suspenders since the watch should be getting updates to the distance and heart rate and also checks then to see if it should kill the workout session.